### ISSUE TITLE
queue.max_bytes isn't respected at start up time

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#6625]
During the startup of logstash the head page is beheaded if it contains unacked events and a new page is created.

This page creation happens even if `queue.max_bytes` has been reached. In the worst case, it's possible to fill up the disk even if queue.max_bytes is 5mb.

Steps to reproduce:

logstash.yml:

```yml
queue.type: persisted
queue.page_capacity: 5mb
queue.max_bytes: 10mb
```

Use a high throughput pipeline that stalls:

`bin/logstash -e "input { generator {} } filter {sleep { time => 1 } }" -w 1 -b 1`

Let logstash start and kill it. Each time this happens a new page is created:

1.

```
/tmp/logstash-5.1.2 % du -sh data/**/*
 10M	data/queue
4.0K	data/queue/checkpoint.0
4.0K	data/queue/checkpoint.head
5.0M	data/queue/page.0
5.0M	data/queue/page.1
4.0K	data/uuid
```

2.

```
/tmp/logstash-5.1.2 % du -sh data/**/*
 15M	data/queue
4.0K	data/queue/checkpoint.0
4.0K	data/queue/checkpoint.1
4.0K	data/queue/checkpoint.head
5.0M	data/queue/page.0
5.0M	data/queue/page.1
5.0M	data/queue/page.2
4.0K	data/uuid
```

3.

```
/tmp/logstash-5.1.2 % du -sh data/**/*
 20M	data/queue
4.0K	data/queue/checkpoint.0
4.0K	data/queue/checkpoint.1
4.0K	data/queue/checkpoint.2
4.0K	data/queue/checkpoint.head
5.0M	data/queue/page.0
5.0M	data/queue/page.1
5.0M	data/queue/page.2
5.0M	data/queue/page.3
4.0K	data/uuid
```

etc..

**Comment by colinsurprenant** [comment]
interesting. not sure how we should deal with that actually. 
also note that these are sparse files. on MacOS they `du` will reports the same size as `ls` but on Linux `du` will show that they are mostly unallocated on disk. see this comment https://github.com/elastic/logstash/pull/6518#issuecomment-276451360

**Comment by jsvd** [comment]
I guess the two questions are:
1. even if the actual file size is close to 0B and the disk is "full" of them, will other applications see the disk as full? Can this still be a problem?
2. Is there a way to know the real size on macosx and any other OS where this happens?

**Comment by colinsurprenant** [comment]
excellent questions - I really do not know how to correctly deal with disk size - we need to discuss this.

**Comment by colinsurprenant** [comment]
so, this relates to #6998 and #6518 - since the beheading operation at startup *may* create a new page, that new head page will be sparsely allocated and on Linux, `du` will actually show zero byte size, but on OSX it shows `page_capacity` byte size. 

So, as discussed in #6998 I suggest we stay safe and check for the higher between [(`max_bytes` - *current queue size*), `page_capacity`] in free disk space in the bootstrap check.

**Comment by colinsurprenant** [comment]
So #6998 should fix this.

**Comment by colinsurprenant** [comment]
fixed by #8978
